command/oauth/cmd.go

```diff
@@ -404,11 +415,26 @@ func disco(provider string) (map[string]interface{}, error) {
 	return details, err
 }
 
+func (o *oauth) NewServer() (*httptest.Server, error) {
+	l, err := net.Listen("tcp", o.CallbackListener)
+	if err != nil {
+		return nil, err
+	}
+	srv := httptest.NewUnstartedServer(o)
+	srv.Listener.Close()
+	srv.Listener = l
+	srv.Start()
+	return srv, nil
+}
```

Comment:
After removing the default value you should change this method. The easiest thing would be to use `httptest.NewServer` if `o.CallbackListerer` is `""`, this will also take care of weird configurations where tcp6 is the only stack available.

And with a custom address, instead of opening an extra listener because of `httptest.NewUnstartedServer` you can just use define the httptest.Server:
```go
srv := &httptest.Server{
	Listener: l,
	Config:   &http.Server{Handler: o},
}
```